packages/cli/src/WorkflowExecuteAdditionalData.ts

```diff
@@ -181,77 +177,6 @@ export function executeErrorWorkflow(
 	}
 }
 
-/**
- * Prunes Saved Execution which are older than configured.
- * Throttled to be executed just once in configured timeframe.
- * TODO: Consider moving this whole function to the repository or at least the queries
- */
-let throttling = false;
-async function pruneExecutionData(this: WorkflowHooks): Promise<void> {
```

Comment:
I'm wondering about the implications of moving this out of `workflowExecuteAfter`. Before we were asking the DB to hard-delete after every execution with throttling, now we're asking it to soft-delete every 15 minutes and hard-delete every hour. Not quite sure if this will take a higher toll on the DB based on the average usage pattern or if it mostly makes no difference?

Reply:
soft-delete is going to be way less expensive than hard-deletes, as we aren't actually deleting any of the execution data.